src/com/goide/settings/GoPluginSettings.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Mihai Toader, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.settings;
+
+import com.goide.util.GoUtil;
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.components.State;
+import com.intellij.openapi.components.Storage;
+import org.jdom.Element;
+import org.jetbrains.annotations.Nullable;
+
+@State(
+  name = "GoPluginSettings",
+  storages = @Storage(id = "other", file = "$APP_CONFIG$/goplugin.xml")
```

Comment:
I'm sorry, I have no idea how to do that. Please advise.

I've tried to add it as it's done for ```GoProjectLibrariesService``` but we don't have it for ```GoApplicationLibrariesService``` and ```GoModuleLibrariesService```.

All I'm getting is:
```
[   9036]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initPersistentComponent(ComponentStoreImpl.java:235)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initComponent(ComponentStoreImpl.java:73)
	at com.intellij.openapi.application.impl.ApplicationImpl.initializeComponent(ApplicationImpl.java:182)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.initializeInstance(ServiceManagerImpl.java:233)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter$1.compute(ServiceManagerImpl.java:216)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:207)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:245)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:211)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:32)
	at com.goide.settings.GoPluginSettings.getInstance(GoPluginSettings.java:45)
	at com.goide.ui.ProjectConfigurationNotification.<init>(ProjectConfigurationNotification.java:44)
[....]
[   9043]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[   9044]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[   9044]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[   9044]  ERROR - impl.stores.ComponentStoreImpl - Current Command: EditorChange 
[   9045]  ERROR - impl.stores.ComponentStoreImpl - Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$ 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initPersistentComponent(ComponentStoreImpl.java:237)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initComponent(ComponentStoreImpl.java:73)
	at com.intellij.openapi.application.impl.ApplicationImpl.initializeComponent(ApplicationImpl.java:182)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.initializeInstance(ServiceManagerImpl.java:233)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter$1.compute(ServiceManagerImpl.java:216)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:207)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:245)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:211)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:32)
	at com.goide.settings.GoPluginSettings.getInstance(GoPluginSettings.java:45)
	at com.goide.ui.ProjectConfigurationNotification.<init>(ProjectConfigurationNotification.java:44)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[....]
[   9047]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  10124]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:144)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
	at com.intellij.ide.SaveAndSyncHandlerImpl$4.onFrameDeactivated(SaveAndSyncHandlerImpl.java:109)
[...]
[  10125]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[  10126]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[  10126]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  10126]   WARN - mponents.impl.stores.StoreUtil - Save settings failed 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl$StateStorageManagerExternalizationSession.setState(StateStorageManagerImpl.java:303)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:145)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
[....]
[  48807]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:144)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
	at com.intellij.ide.SaveAndSyncHandlerImpl$4.onFrameDeactivated(SaveAndSyncHandlerImpl.java:109)
	at com.intellij.ide.FrameStateManagerImpl.fireDeactivationEvent(FrameStateManagerImpl.java:87)
[....]
[  48808]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  48808]   WARN - mponents.impl.stores.StoreUtil - Save settings failed 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl$StateStorageManagerExternalizationSession.setState(StateStorageManagerImpl.java:303)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:145)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
[....]
```